roscala/src/main/scala/coop/rchain/rosette/Location.scala

```diff
@@ -1,406 +1,94 @@
-            if (argReg < k.argvec.elem.size) {
-              k.argvec.elem(argReg)
-            } else {
-              Ob.INVALID
-            }
+        case BitField(indirect, level, offset, spanSize, sign) => pure(None)
 
-          case LTLexVariable(indirect, level, offset) =>
-            k.env.getLex(indirect, level, offset)
+        case BitField00(offset, spanSize, sign) => pure(None)
 
-          case LTAddrVariable(indirect, level, offset) =>
-            k.env.getAddr(indirect, level, offset)
-
-          case LTGlobalVariable(offset) =>
-            if (offset < globalEnv.numberOfSlots()) {
-              globalEnv.slot(offset)
-            } else {
-              Ob.INVALID
-            }
-
-          case LTBitField(indirect, level, offset, spanSize, sign) =>
-            k.env.getField(indirect, level, offset, spanSize, sign)
-
-          case LTBitField00(offset, spanSize, sign) =>
-            k.env.getField(false, 0, offset, spanSize, sign)
-
-          case LTLimbo => Ob.INVALID
-        }
-    }
-
-  def printRep(loc: Location): String = {
-    val names = List("rslt",
-                     "trgt",
-                     "argvec",
-                     "env",
-                     "code",
-                     "ctxt",
-                     "self",
-                     "self-env",
-                     "rcvr",
-                     "monitor")
-
-    loc match {
-      case LocationAtom(_) => {
-        suicide("Location.printRep: not a generic type")
-        ""
+        // TODO:
+        case _ => pure(None)
       }
-
-      case LocationGT(genericType) =>
-        genericType match {
-          case LTCtxtRegister(reg) =>
-            if (0 <= reg && reg < NumberOfCtxtRegs) {
-              names(reg)
-            } else {
-              f"unknown ctxt register 0x$reg%x"
-            }
-
-          case LTArgRegister(argReg) =>
-            s"arg[$argReg]"
-
-          case LTLexVariable(indirect, level, offset) => {
-            val offsetStr = if (indirect) s"($offset)" else s"$offset"
-            s"lex[$level,$offsetStr]"
-          }
-
-          case LTAddrVariable(indirect, level, offset) => {
-            val offsetStr = if (indirect) s"($offset)" else s"$offset"
-            s"addr[$level,$offsetStr]"
-          }
-
-          case LTGlobalVariable(offset) =>
-            s"global[$offset]"
-
-          case LTBitField(indirect, level, offset, spanSize, sign) => {
-            val signStr = if (sign != 0) "s" else "u"
-            val offsetStr = if (indirect) s"($offset)" else s"$offset"
-            s"${signStr}fld[$level,$offsetStr,$spanSize]"
-          }
-
-          case LTBitField00(offset, spanSize, sign) => {
-            val signStr = if (sign != 0) "s" else "u"
-            s"${signStr}fld[$offset,$spanSize]"
-          }
-
-          case LTLimbo => "limbo"
-        }
-    }
+    } yield res
   }
 
-  def valWRT(loc: Location, v: Ob, globalEnv: TblObject): Ob =
```

Comment:
Do you have some problems adapting `valWrt` implementation?

Reply:
Not really.
Just the fact that this rewrite is blocking other things and if I implement it, I want to do it right (with tests and observing it in gdb for which I don't have time at the moment - we should put the ticket back to the backlog instead of marking it as done in Jira).